model/span.go

```diff
@@ -28,6 +28,25 @@ var (
 	ErrValidDurationRequired = errors.New("valid duration required")
 )
 
+// Baggage holds the interface for interacting with baggage context propagation
+// implementations.
+type Baggage interface {
+	// Init returns a fresh Baggage primed for usage in a request lifecycle.
+	Init() Baggage
+	// AddHeader adds the provided values to a header designated by key. If key
+	// is not in the whitelist it will be a noop and return false.
+	AddHeader(key string, value ...string) bool
+	// DeleteHeader removes the header data designated by key. If key is not
+	// in the whitelist it will be a noop and return false.
+	DeleteHeader(key string) bool
+	// IterateHeaders will iterate over the whitelist headers and for each found
```

Comment:
I'd suggest whitelist is an internal detail as there's no api to change it and wouldn't likely occur here anyway. Maybe just remove mention to that?  IterateWhiteList -> IterateKeys also?